Mathlib/Topology/Algebra/Module/Basic.lean

```diff
@@ -2272,34 +2272,153 @@ end ContinuousLinearEquiv
 
 namespace ContinuousLinearMap
 
-variable {R : Type*} {M : Type*} {M₂ : Type*} [TopologicalSpace M] [TopologicalSpace M₂]
+variable {R : Type*} {M M₂ M₃ : Type*}
+  [TopologicalSpace M] [TopologicalSpace M₂] [TopologicalSpace M₃]
 
 section
 
 variable [Semiring R]
-variable [AddCommMonoid M₂] [Module R M₂]
-variable [AddCommMonoid M] [Module R M]
+  [AddCommMonoid M] [Module R M]
+  [AddCommMonoid M₂] [Module R M₂]
+  [AddCommMonoid M₃] [Module R M₃]
+
+/-- A continuous linear map is invertible if it is the forward direction of a continuous linear
+equivalence. -/
+def IsInvertible (f : M →L[R] M₂) : Prop :=
+  ∃ (A : M ≃L[R] M₂), A = f
```

Comment:
Is this the kind of thing we *only* need for `ContinuousLinearMap`? I understand that for a lot of algebraic morphisms it can be phrased in terms of `Function.Bijective f`, but if this is a pattern that we might need more frequently for other morphism classes, then I wonder if it would be better to have some sort of generic machinery for `FunLike F R S` and `EquivLike E R S` that says:
```
def IsInvertible {F : Type*} {E R S : outParam (Type*)} (f : F) :=
   ∃ (e : E), ⇑e = f
```
Maybe it doesn't buy us anything because we have no `symm` or `trans` for `EquivLike`. Probably this comment is worthless.

Reply:
Yes, I wondered the same. But I don't think the design of `EquivLike` makes it possible to have something like that in a usable way, because of the lack of `symm`. I have to admit that, while I love `FunLike`, I am not completely convinced by `EquivLike` because of these shortcomings...